### PR TITLE
MyPlants: User can now addplants

### PR DESCRIPTION
**Pull request by AronSeamountain**
Co-authored-by: ilcarolinejonsson <ilcarolinejonsson@gmail.com>

### CODE REVIEW

**Comment by AronSeamountain** [comment]
Added functionality to add a plant.

File: blomsterLandet/src/screens/shop/ShopItemPreviewScreen.js [hunk 1]
```diff
@@ -14,15 +14,17 @@ class ShopItemPreviewScreen extends Component {
 
     componentWillMount() {
         self = this;
-        image = images[this.props.navigation.getParam('product').key];
+        console.log(this.props.navigation.getParam('product').name);
+        console.log("HÄÄÄÄÄÄR");
```
Comment (hagerg) [inline blomsterLandet/src/screens/shop/ShopItemPreviewScreen.js]: Console.log

File: blomsterLandet/src/screens/shop/ShopItemPreviewScreen.js [hunk 2]
```diff
@@ -14,15 +14,17 @@ class ShopItemPreviewScreen extends Component {
 
     componentWillMount() {
         self = this;
-        image = images[this.props.navigation.getParam('product').key];
+        console.log(this.props.navigation.getParam('product').name);
```
Comment (hagerg) [inline blomsterLandet/src/screens/shop/ShopItemPreviewScreen.js]: Console.log

File: blomsterLandet/src/services/plantHandlerHelperFunctions.js [hunk 3]
```diff
@@ -46,6 +46,22 @@ export function waterPlant(db, plant) {
...
+    let newIndex;
+    console.log("IN PLANTHELPER")
+    if (db.plantList && db.plantList.length > 0) {
+        newIndex = db.plantList[db.plantList.length - 1].key + 1;
+    } else {
+        newIndex = 0;
+    }
+    console.log("new Index: " + newIndex);
+    plant.key = newIndex;
+    console.log(db.plantList);
```
Comment (hagerg) [inline blomsterLandet/src/services/plantHandlerHelperFunctions.js]: console.log

File: blomsterLandet/src/services/plantHandlerHelperFunctions.js [hunk 4]
```diff
@@ -46,6 +46,22 @@ export function waterPlant(db, plant) {
...
+    db.plantList.push(plant);
+    console.log(db.plantList);
```
Comment (hagerg) [inline blomsterLandet/src/services/plantHandlerHelperFunctions.js]: console.log

File: blomsterLandet/src/services/plantHandlerHelperFunctions.js [hunk 5]
```diff
@@ -46,6 +46,22 @@ export function waterPlant(db, plant) {
...
 
+export function addPlant(db, plant) {
+    let newIndex;
+    console.log("IN PLANTHELPER")
+    if (db.plantList && db.plantList.length > 0) {
+        newIndex = db.plantList[db.plantList.length - 1].key + 1;
+    } else {
+        newIndex = 0;
+    }
+    console.log("new Index: " + newIndex);
```
Comment (hagerg) [inline blomsterLandet/src/services/plantHandlerHelperFunctions.js]: console.log

File: blomsterLandet/src/services/plantHandlerHelperFunctions.js [hunk 6]
```diff
@@ -46,6 +46,22 @@ export function waterPlant(db, plant) {
     console.log("could not find plant key in database");
 }
 
+export function addPlant(db, plant) {
+    let newIndex;
+    console.log("IN PLANTHELPER")
```
Comment (hagerg) [inline blomsterLandet/src/services/plantHandlerHelperFunctions.js]: console.log

File: blomsterLandet/src/screens/myplants/AddPlantScreen.js [hunk 7]
```diff
@@ -4,89 +4,87 @@ import { connect } from 'react-redux';
...
+            lastWatered: Date.now() / 1000,
+            wateringInterval: wI
+        };
+        self.props.addPlant(newPlant);
+        self.props.navigation.navigate('Home');
+        plantHandler.createFile(self.props.myPlants);
+    }
 
     renderPlantTypes() {
         console.log('renderPlantTypes');
```
Comment (hagerg) [inline blomsterLandet/src/screens/myplants/AddPlantScreen.js]: console.log

**Pull request merged**